azurerm/resource_arm_front_door.go

```diff
@@ -0,0 +1,1460 @@
+										Required: true,
+									},
+								},
+							},
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.BackendPoolRoutingRuleName,
+						},
+						"health_probe_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"load_balancing_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"frontend_endpoint": {
+				Type:     schema.TypeList,
+				MaxItems: 100,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"name": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validate.BackendPoolRoutingRuleName,
+						},
+						"host_name": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"session_affinity_enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"session_affinity_ttl_seconds": {
+							Type:     schema.TypeInt,
+							Optional: true,
+							Default:  0,
+						},
+						"custom_https_provisioning_enabled": {
+							Type:     schema.TypeBool,
+							Required: true,
+						},
+						"custom_https_configuration": {
+							Type:     schema.TypeList,
+							Optional: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"certificate_source": {
+										Type:     schema.TypeString,
+										Optional: true,
+										ValidateFunc: validation.StringInSlice([]string{
+											string(frontdoor.CertificateSourceAzureKeyVault),
+											string(frontdoor.CertificateSourceFrontDoor),
+										}, false),
+										Default: string(frontdoor.CertificateSourceFrontDoor),
+									},
+									"provisioning_state": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									"provisioning_substate": {
+										Type:     schema.TypeString,
+										Computed: true,
+									},
+									// NOTE: None of these attributes are valid if
+									//       certificate_source is set to FrontDoor
+									"azure_key_vault_certificate_secret_name": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"azure_key_vault_certificate_secret_version": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"azure_key_vault_certificate_vault_id": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"tags": tagsSchema(),
+		},
+
+		CustomizeDiff: func(d *schema.ResourceDiff, v interface{}) error {
+			if err := validate.FrontdoorSettings(d); err != nil {
```

Comment:
Customize diff doesn't work for interpolated values, so we should move this to create.

Reply:
If I move it to create, for this particular resource it makes the user experience very poor. I would like to keep the validation running in the customize diff if possible.

Reply:
👍 just be aware it might break and behave in strange and wonderful ways

Reply:
If this causes issues, I will be happy to take full blame, as it was deserved and my choice to take this course of action.